app/controllers/concerns/filter_by_owner.rb

```diff
@@ -10,11 +10,9 @@ def add_owner_ids_to_filter_param!
     unless owner_filter.blank?
       groups = UserGroup.where(UserGroup.arel_table[:name].lower.in(owner_filter.map(&:downcase)))
       @controlled_resources = controlled_resources
-        .eager_load(:owner)
-        .joins(:owner)
-        .joins(:access_control_lists)
-        .where(access_control_lists: {user_group: groups})
-        .where.overlap(access_control_lists: { roles: ["owner"] })
+      .eager_load(owner: [:users])
+      .joins(:owner)
+      .where(access_control_lists: { user_group: groups })
```

Comment:
So you removed the join on access_control_lists, but you're still able to query on it. How does that work? And how can we make that more obvious in the code?